src/verifier.cc

```diff
@@ -25,29 +25,28 @@
 namespace amber {
 namespace {
 
-const uint8_t kBitsPerByte = 8;
+const uint32_t kBitsPerByte = 8;
 const double kEpsilon = 0.000001;
 const double kDefaultTexelTolerance = 0.002;
 
 // Copy [src_bit_offset, src_bit_offset + bits) bits of |src| to
 // [0, bits) of |dst|.
 void CopyBitsOfMemoryToBuffer(uint8_t* dst,
                               const uint8_t* src,
-                              uint8_t src_bit_offset,
+                              uint32_t src_bit_offset,
                               uint32_t bits) {
-  while (src_bit_offset > static_cast<uint8_t>(7)) {
+  while (src_bit_offset > 7) {
     ++src;
-    src_bit_offset = static_cast<uint8_t>(src_bit_offset - kBitsPerByte);
+    src_bit_offset = src_bit_offset - kBitsPerByte;
   }
 
   // Number of bytes greater than or equal to |(src_bit_offset + bits) / 8|.
-  const uint8_t size_in_bytes =
-      static_cast<uint8_t>((src_bit_offset + bits + 7) / kBitsPerByte);
-  assert(size_in_bytes <= static_cast<uint8_t>(kBitsPerByte));
+  const uint32_t size_in_bytes = (src_bit_offset + bits + 7) / kBitsPerByte;
```

Comment:
These were all casted correctly, what was the error from CTS?

Reply:
```
[ 34%] Building C object framework/qphelper/CMakeFiles/qphelper.dir/qpInfo.c.o
/home/ari/git/issueNewAmber/vk-gl-cts/external/amber/src/src/verifier.cc: In function ‘std::vector<double> amber::{anonymous}::GetActualValuesFromTexel(const uint8_t*, const amber::Format*)’:
/home/ari/git/issueNewAmber/vk-gl-cts/external/amber/src/src/verifier.cc:271:36: error: conversion from ‘uint32_t’ {aka ‘unsigned int’} to ‘uint8_t’ {aka ‘unsigned char’} may change value [-Werror=conversion]
       bit_offset += seg.GetNumBits();
                                    ^
/home/ari/git/issueNewAmber/vk-gl-cts/external/amber/src/src/verifier.cc:323:16: error: conversion from ‘uint32_t’ {aka ‘unsigned int’} to ‘uint8_t’ {aka ‘unsigned char’} may change value [-Werror=conversion]
     bit_offset += num_bits;
     ~~~~~~~~~~~^~~~~~~~~~~
```